python/cudf/dataframe/dataframe.py

```diff
@@ -565,7 +565,8 @@ def add_column(self, name, data, forceindex=False):
         if name in self._cols:
             raise NameError('duplicated column name {!r}'.format(name))
 
-        series = Series(data)
+        if isinstance(data, type(1 for i in range(1))):
+            series = Series(data)
```

Comment:
You need an else here to initialize `series`. This is the cause of your current test failures.

Reply:
Please run the tests locally -- running pytests locally should have tripped this before you pushed.